Documentation/advanced-configuration.md

```diff
@@ -283,129 +289,90 @@ ceph osd pool set rbd pg_num 512
-```bash
-# Create a ConfigMap from a text file
-kubectl -n rook create configmap rook-config --from-file ceph.conf
-
-# View the resulting ConfigMap
-kubectl -n rook get configmap rook-config -o yaml
+When the Rook Operator creates a cluster, a placeholder ConfigMap is created that
+will allow you to override Ceph configuration settings. There are four categories
+of settings, one for each daemon. When the daemon pods are started, the 
+settings specified in this ConfigMap will be merged with the default settings
+generated by Rook. 
+
+The default override settings are blank. Cutting out the extraneous properties, 
+we would see the following defaults after creating a cluster:
+```
+$ kubectl -n rook get ConfigMap config-override -o yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: config-override
+  namespace: rook
+data:
+  mds: ""
+  mon: ""
+  osd: ""
+  rgw: ""
 ```
 
-If you want to make changes you can edit the ConfigMap within the Kubernetes
-cluster:
+To apply your desired configuration, you will need to update this ConfigMap.
+The next time the daemon pod(s) start, the settings will be merged with the default
+settings created by Rook.
+
 ```bash
-kubectl -n rook edit configmap rook-config
+kubectl -n rook edit configmap config-override
 ```
 
-Alternatively you can maintain a copy of the ConfigMap on your local computer
-that looks like this:
+Modify the settings and save. Each line you add should be indented from the daemon name as such:
 ```yaml
-# Filename: rook-config.yaml
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: rook-config
+  name: config-override
+  namespace: rook
 data:
-  ceph.conf: |
+  mds: ""
+  mon: ""
+  osd: |
     [global]
     osd crush update on start = false
     osd pool default size     = 2
+  rgw: ""
 ```
 
-Then to create or apply changes to the ConfigMap run:
-```bash
-kubectl -n rook apply -f ./rook-config.yaml
-```
-
-#### Modify The `osd` DaemonSet
-
-We will tell the OSD pods to mount our rook-config/ceph.conf as a file and add
-the `--ceph-config-override` argument pointing to that file.
-
-Start by editing the `osd` DaemonSet:
-```bash
-kubectl -n rook edit ds osd
-```
-
-You'll need to add a `volume` and `volumeMount` for the rook-config, so those
-sections might look like this to make /etc/rook/ceph.conf available in the pods:
-```yaml
-        volumeMounts:
-          - mountPath: /var/lib/rook
-            name: rook-data
-            readOnly: false
-          - mountPath: /dev
-            name: devices
-          - mountPath: /etc/rook
-            name: rook-config
-            readOnly: true
-```
-```yaml
-      volumes:
-        - name: rook-data
-          hostPath:
-            path: /var/lib/rook
-        - hostPath:
-            path: /dev
-          name: devices
-        - name: rook-config
-          configMap:
-            name: rook-config
-```
+Each daemon will need to be restarted where you want the settings applied, although
```

Comment:
Should we make this more automatic? Can we have a watcher for that config-override configmap and have it restart the corresponding daemons based on the config? That would be a much better experience.